File: 2020-03/3-19/409最长回文串/409.py


class Solution:
    def longestPalindrome(self, s: str) -> int:
        record = {}
        res = 0
        for elem in s:
            record[elem] = record.get(elem, 0) + 1
            if record[elem] % 2 == 0:
                res += 2
        return res if res == len(s) else res + 1

File: 2020-05/5-04/409最长回文串/409.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char, int> record;
        int res = 0;

        for(auto c: s)
            record[c]++;

        for(auto c: record)
            res += c.second / 2 * 2;

        return res == s.size() ? res : res + 1;
    }
};